src/workerd/jsg/util.c++

```diff
@@ -806,4 +806,29 @@ v8::MaybeLocal<v8::String> newExternalTwoByteString(Lock& js, kj::ArrayPtr<const
   return ExternTwoByteString::createExtern(js.v8Isolate, buf);
 }
 
+void captureStackTrace(v8::Isolate* isolate, v8::Local<v8::Object> target) {
+  auto context = isolate->GetCurrentContext();
+  auto global = context->Global();
+
+  auto name = v8StrIntern(isolate, "captureStackTrace");
+  // This is a bit of a hacky trick. The context has this additional "extra bindings object"
+  // attached to it that v8 uses to expose some built ins. For instance, it is used by Node.js
+  // to get access to the Trace and IsTraceCategoryEnabled built-in functions. We use it here
+  // to cache the reference to the Error.captureStackTrace function.
+  auto extras = context->GetExtrasBindingObject();
+  v8::Local<v8::Value> maybeCaptureFn = check(extras->Get(context, name));
+  if (maybeCaptureFn->IsUndefined()) {
+    auto error = check(global->Get(context, v8StrIntern(isolate, "Error")));
+    JSG_REQUIRE(error->IsFunction(), TypeError, "The Error global is not a constructor.");
+    auto capture = check(error.As<v8::Object>()->Get(context, name));
```

Comment:
What if the application monkey-patches this? Could it cause problems for us? Perhaps we should capture this handle at startup before executing application code?

Reply:
Or, patch v8 to expose it here itself so we don't have to do this dance.

Reply:
Capturing this function in `ResourceWrapper::newContext()`, or a V8 patch, both sound fine to me.